### PR TITLE
Fix form property assignment edge case

### PR DESCRIPTION
**Pull request by wxiaoguang**
"form" has an edge case: its `<input name=action>` element overwrites the `action` property, we can only set attribute.

This PR makes `assignElementProperty` can handle such case, and add more tests

### CODE REVIEW

File: web_src/js/features/common-button.ts [hunk 1]
```diff
@@ -109,18 +109,26 @@ function onHidePanelClick(el: HTMLElement, e: MouseEvent) {
...
 
-export function assignElementProperty(el: any, name: string, val: string) {
-  name = camelize(name);
-  const old = el[name];
+export type ElementWithAssignableProperties = {
+  getAttribute: (name: string) => string | null;
+  setAttribute: (name: string, value: string) => void;
+} & Record<string, any>
+
+export function assignElementProperty(el: ElementWithAssignableProperties, kebabName: string, val: string) {
```
Comment (silverwind) [inline web_src/js/features/common-button.ts:117]: Why not `Element` or `HTMLElement`?

Comment (wxiaoguang) [reply web_src/js/features/common-button.ts:117]: Because it needs "any":

Image

Comment (silverwind) [reply web_src/js/features/common-button.ts:117]: Shouldn't `keyof typeof Element` work for the property access?

```ts
  const camelizedName: keyof typeof Element = camelize(kebabName);

Comment (silverwind) [reply web_src/js/features/common-button.ts:117]: If not, use `(el as any)[camelizedName] = ...` to assign. Better do dirty stuff within a function than to expose `any` to consumers.

Comment (wxiaoguang) [reply web_src/js/features/common-button.ts:117]: I don't see a clearer solution.

Feel free to edit this PR directly with your solution.

**Review by delvh — APPROVED** [review]

**Review by lunny — APPROVED** [review]
